JsonLogic/Rule.cs

```diff
@@ -119,9 +111,11 @@ public override Rule Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSer
 				if (ruleType == null)
 					throw new JsonException($"Cannot identify rule for {op}");
 
+				var typeInfo = LogicSerializerContext.OptionsManager.SerializerOptions.GetTypeInfo(ruleType);
```

Comment:
Let's just go straight to the RuleRegistry's lookup table and then we don't need to go through the ExternalTypeInfoResolvers thing. We might even be able to get rid of the OptionsManager.

Reply:
I'll give that a go.